master/buildbot/steps/source/git.py

```diff
@@ -159,6 +159,9 @@ def __init__(self, repourl=None, branch='HEAD', mode='incremental', method=None,
     def startVC(self, branch, revision, patch):
         self.branch = branch or 'HEAD'
         self.revision = revision
+        # ignore the revision if the branch ends with /merge
+        if self.branch.endswith("/merge"):
```

Comment:
@sa2ajj I think I addressed your request. I need your approval to merge as you expressly required changes